1.x.x release blog post (#174)
MDAnalysis/MDAnalysis.github.io: _posts/2021-05-02-release-1.x.x.md

---
layout: post
title: Release 1.0.0, 1.0.1, 1.1.1
---

A new release of MDAnalysis now available!
Or rather, three releases have been made for the 1.x version of MDAnalysis, so we just talk about all of them in a single post.

This version brings a multitude of **fixes**, [**deprecations**](#deprecations), and **new features**.

> Some highlights are given below, whilst the release notes(someone make this a link to the right place please) list all the changes in this version.

Reviewer: ```suggestion
Some highlights are given below, whilst the [CHANGELOG](https://github.com/MDAnalysis/mdanalysis/blob/release-1.1.1/package/CHANGELOG).
```

We haven't had wiki-based release notes since 0.19.2.

This release completes, albeit with a small delay, the first phase of our [roadmap][]. It is the last version which will support python 2.7 and 3.5. Our next release, 2.0.0, which will be made available soon, will include major API changes and will support python 3.6+

## GSoC? 

## Notable new additions

> 

Reviewer: Stable API (for 1.x). Semantic versioning guarantees no backward incompatible changes within 1.x

Reviewer 2: I mentioned this above, does that suffice?

Reviewer: yes

- Writer.write_next_timestep() is deprecated in favor of Writer.write().
- Passing Timestep objects to Writer.write() is deprecated. In 2.0.0 only Universe or AtomGroup objects will be accepted.
- The way in which the NCDFWriter handles scale factos will change in version 2.x (see Issue #2327 for more details).
- When writing PDB files, MDAnalysis will no longer be using the last letter of the SegID to set the chainID in version 2.0.0.
- The bfactors and tempfactors attributes (set by the PDB and MMTF parsers respectively), will be aliased in version 2.0.0.
- When parsing TPR files, resids will be indexed from 1 rather than the current default of 0.

The following part of the core and library components will be removed/changed in version 2.0.0:
- lib.log.echo() is deprecated in favor of the new lig.log.ProgressBar.
- core.universe.as_Universe() is deprecated.

## Author statistics

Altogether this represents the work of 42 contributors from around the world,
and featured the work of 25 new contributors:
Note: here is a list of all the new contributors' github handles, can someone please convert this to a list with their names and a link to their github profiles? i.e.:

>  - [Irfan Alibay][]

Reviewer: Why not 
```
Irfan Alibay (@IAlibay)
```
or just the GH handles?

We can always link to AUTHORS.

Author: I was just imitating the 0.19.x blog post, whatever is easier is best here.

---
layout: post
title: Version 1
---

> We are pleased to release version 1 of MDAnalysis!

Reviewer 2: > pleased

Feel free to substitute a more excited word.

MDAnalysis 1.x is a stable legacy platform that supports Python 2.7, 3.5, 3.6, 3.7 and 3.8 on Windows (32- and 64-bit), MacOS, and Linux.
The API will not change for any upcoming 1.x versions.
This release completes the first phase of our [roadmap][].
From here on, we will focus on developing version 2 of MDAnalysis, which will support Python 3.6+ only, and will include major API changes and improvements.

The 1.0.0, 1.0.1 and 1.1.1 versions of MDAnalysis are the product of more than 18 months effort.
They contain a multitude of **fixes**, [**deprecations**](#deprecations), and **new features**.
We highlight important changes below, but users are strongly encouraged to read the [CHANGELOG][] for full details.
All users are recommended to upgrade to 1.1.1, as it includes important fixes to neighbor searching, a fundamental component of many analyses in MDAnalysis.

In [Dependencies](#dependencies) we list the required dependencies of MDAnaysis version 1.

> ## GSoC? 

Reviewer 2: @IAlibay not sure what you wanted to say here, aren't most of the changes in 2.0?

Author: This is following the 0.19.x format, the question here is "are there GSOC contributions we want to highlight". I guess here we are mainly thinking about GSOC 2019 or GSOD?

    - a method to compute the root-mean-square-inner-product of subspaces
    - a method to calculate the cumulative overlap of a vector in a subspace

Other additions to core functionality include:
    - [wrap and unwrap on-the-fly transformations](https://docs.mdanalysis.org/stable/documentation_pages/transformations/wrap.html)
    - methods to add [bonds][add_bonds], [angles][add_angles], [dihedrals][add_dihedrals], and [impropers][add_impropers] to a Universe
    - methods to delete [bonds][del_bonds], [angles][del_angles], [dihedrals][del_dihedrals], and [impropers][del_impropers] from a Universe
    - the [get_connections][] method to get bonds, angles, dihedrals and impropers in an AtomGroup, with the ability to specify whether all atoms involved is within the AtomGroup (`outside=False`), or whether *at least one* atom involved is within the AtomGroup (`outside=True`)

Miscellaneous performance improvements include:
    - Dihedral selection in the Ramachandran class has been sped up ~700x.
    - TPR parsing has been sped up 2–30x.

## Support for Windows 32

> - Add some blurb here

Reviewer 2: @IAlibay I'm not really sure what to say here other than "we added support for Windows 32-bit", which is up top.

Author: Not as necessary heading, we can just put it down as a bullet point under enhancements

 - [@AnshulAngaria](https://github.com/AnshulAngaria)
 - [@ss62171](https://github.com/ss62171)
 - [@Luthaf](https://github.com/Luthaf)
 - [@yuxuanzhuang](https://github.com/yuxuanzhuang)
 - [@abhishandy](https://github.com/abhishandy)
 - [@mlnance](https://github.com/mlnance)
 - [@shfrz](https://github.com/shfrz)
 - [@wvandertoorn](https://github.com/wvandertoorn)
 - [@cbouy](https://github.com/cbouy)
 - [@AmeyaHarmalkar](https://github.com/AmeyaHarmalkar)
 - [@Oscuro-Phoenix](https://github.com/Oscuro-Phoenix)
 - [@andrrizzi](https://github.com/andrrizzi)
 - [@WG150](https://github.com/WG150)
 - [@Marcello-Sega](https://github.com/Marcello-Sega)
 - [@calcraven](https://github.com/calcraven)
 - [@ianmkenney](https://github.com/ianmkenney)
 - [@rcrehuet](https://github.com/rcrehuet)
 - [@hanatok](https://github.com/hanatok)

## Upgrading to MDAnalysis version 1.1.1

To get all these features run either `conda update -c conda-forge mdanalysis`
or `pip install --upgrade MDAnalysis`.

## Dependencies

We list below the core dependencies and versions that MDAnalysis has been tested on.
They are provided as a string for easy use with conda or pip:

>  - Python 2.7: biopython==1.76 cython==0.29.15 griddataformats==0.5.0 gsd==1.7.0 matplotlib==2.2.5 mmtf-python==1.1.2 netcdf4==1.3.1 numpy==1.16.5 scipy==1.2.1 tqdm==4.60.0

Reviewer 2: @IAlibay is this what you meant by requirements.txt? I'm not totally sure what listing the max packages does without listing the Python version to go with.

Author: Sure, depends on everyone else as to whether it's the right place to put this information but it seems reasonable to me.

Reviewer: looks good to me, especially for a legacy release

---
layout: post
title: Version 1
---

We are happy to release version 1 of MDAnalysis!

MDAnalysis 1.x is a stable legacy platform that supports 

> -  Python 2.7 (on Linux and MacOS only), 

Reviewer: If we do not have conda packages for 2.7 then we need to say so somewhere, perhaps link to issue https://github.com/MDAnalysis/mdanalysis/issues/3105

> -  Python 3.5, 3.6, 3.7 and 3.8 (on Windows (32- and 64-bit), MacOS, and Linux).

Reviewer: please check: no Windows for 2.7, right?

Author: Yup, 3.5+ for Windows.

The API will not change for any upcoming 1.x versions.
This release completes the first phase of our [roadmap][].
From here on, we will focus on developing version 2 of MDAnalysis, which will support Python 3.6+ only, and will include major API changes and improvements.

The 1.0.0, 1.0.1 and 1.1.1 versions of MDAnalysis are the product of more than 18 months effort.
They contain a multitude of **fixes**, [**deprecations**](#deprecations), and **new features**.
We highlight important changes below, but users are strongly encouraged to read the [CHANGELOG][] for full details.
All users are recommended to upgrade to 1.1.1, as it includes important fixes to neighbor searching, a fundamental component of many analyses in MDAnalysis.

In [Dependencies](#dependencies) we list the required dependencies of MDAnaysis version 1.

## Notable new additions

We now support several new formats:
- [FHI-aims](https://userguide.mdanalysis.org/stable/formats/reference/in)
- [NAMD binary coordinates](https://userguide.mdanalysis.org/stable/formats/reference/coor)
- [ITP files from GROMACS](https://userguide.mdanalysis.org/stable/formats/reference/itp)

> We have also added [*converters*](https://docs.mdanalysis.org/stable/documentation_pages/converters) to and from other popular analysis packages.

Author: ```suggestion
We have also added [*converters*](https://docs.mdanalysis.org/1.1.1/documentation_pages/converters) to and from other popular analysis packages.
```

Where possible I think we should be linking to 1.1.1 not stable. That way when 2.0 gets released these links don't suddenly start pointing at the wrong place?

@lilyminium I'll default to whatever you think is best here, also is there a way to do something similar for the userguide?

Author: merging these after discussion with @lilyminium on discord

We plan to expand in this exciting direction in future versions, as laid out in our [interoperability][] roadmap.
For now, we support:
- [ParmEd objects](https://userguide.mdanalysis.org/stable/formats/reference/parmed)
- [the chemfiles library](https://userguide.mdanalysis.org/stable/formats/reference/chemfiles)

New additions to analysis include:
- `frames` and `times` attributes to AnalysisBase to capture the frames and times that the analysis was `run()` on. This is accessible to all analyses that subclass AnalysisBase.

> - a [correlations module](https://docs.mdanalysis.org/stable/documentation_pages/lib/correlations.html) for computing the discrete autocorrelation function

Author: ```suggestion
- a [correlations module](https://docs.mdanalysis.org/1.1.1/documentation_pages/lib/correlations.html) for computing the discrete autocorrelation function
```

> - a [HydrogenBondAnalysis](https://docs.mdanalysis.org/stable/documentation_pages/analysis/hydrogenbonds.html) class for improved and more efficient analysis of hydrogen bonds

Author: ```suggestion
- a new [HydrogenBondAnalysis](https://docs.mdanalysis.org/1.1.1/documentation_pages/analysis/hydrogenbonds.html) class for improved and more efficient analysis of hydrogen bonds, which replaces the now deprecated [hbond_analysis](https://docs.mdanalysis.org/1.1.1/documentation_pages/analysis/hbond_analysis.html) code
```
"new" isn't really helping here, but I'm thinking somehow we need to make it clear that it's a replacement for old code?

> - an [AverageStructure](https://docs.mdanalysis.org/stable/documentation_pages/analysis/align.html#MDAnalysis.analysis.align.AverageStructure) class for computing the average structure of a trajectory out of memory

Author: "out of memory" here might be a little bit cryptic for unaware readers?

> - an [AverageStructure](https://docs.mdanalysis.org/stable/documentation_pages/analysis/align.html#MDAnalysis.analysis.align.AverageStructure) class for computing the average structure of a trajectory out of memory

Author: ```suggestion
- an [AverageStructure](https://docs.mdanalysis.org/1.1.1/documentation_pages/analysis/align.html#MDAnalysis.analysis.align.AverageStructure) class for computing the average structure of a trajectory out of memory
```

> - a [hole2](https://docs.mdanalysis.org/stable/documentation_pages/analysis/hole2.html) module for improved interfacing with the HOLE2 program

Author: ```suggestion
- a [hole2](https://docs.mdanalysis.org/1.1.1/documentation_pages/analysis/hole2.html) module for improved interfacing with the HOLE2 program, which replaces the now deprecated [hole](https://docs.mdanalysis.org/1.1.1/documentation_pages/analysis/hole.html) module
```

> - a [DensityAnalysis](https://docs.mdanalysis.org/stable/documentation_pages/analysis/density.html#MDAnalysis.analysis.density.DensityAnalysis) class for improved density analysis

Author: ```suggestion
- a [DensityAnalysis](https://docs.mdanalysis.org/1.1.1/documentation_pages/analysis/density.html#MDAnalysis.analysis.density.DensityAnalysis) class for improved density analysis, replacing the now deprecated [density_from_Universe()](https://docs.mdanalysis.org/1.1.1/documentation_pages/analysis/density.html#MDAnalysis.analysis.density.density_from_Universe) code
```

- a method to compute the root-mean-square-inner-product of subspaces
- a method to calculate the cumulative overlap of a vector in a subspace

Other additions to core functionality include:

> - [wrap and unwrap on-the-fly transformations](https://docs.mdanalysis.org/stable/documentation_pages/transformations/wrap.html)

Author: ```suggestion
- [wrap and unwrap on-the-fly transformations](https://docs.mdanalysis.org/1.1.1/documentation_pages/transformations/wrap.html)
```

- methods to add [bonds][add_bonds], [angles][add_angles], [dihedrals][add_dihedrals], and [impropers][add_impropers] to a Universe
- methods to delete [bonds][del_bonds], [angles][del_angles], [dihedrals][del_dihedrals], and [impropers][del_impropers] from a Universe
- the [get_connections][] method to get bonds, angles, dihedrals and impropers in an AtomGroup, with the ability to specify whether all atoms involved is within the AtomGroup (`outside=False`), or whether *at least one* atom involved is within the AtomGroup (`outside=True`)

Miscellaneous performance improvements include:
- Dihedral selection in the Ramachandran class has been sped up ~700x.
- TPR parsing has been sped up 2–30x.

## Notable improvements

> We have added significant flexibility to our atom selection language.

Reviewer: examples? Also `<pet peeve>`I dislike use of  "significant" when not supported by some sort of error analysis`</pet peeve>`.

> We have added significant flexibility to our atom selection language.

Author: ```suggestion
We have improved the flexibility to our atom selection language, allowing for advanced pattern matching operators.

For example, we now support the `?` for single character matching, so using `T?R` in a selection string for a protein would yield both `THR` and TYR`. More information can be found in our [selection documentation](https://docs.mdanalysis.org/1.1.1/documentation_pages/selections.html#pattern-matching).
```

> - the [get_connections][] method to get bonds, angles, dihedrals and impropers in an AtomGroup, with the ability to specify whether all atoms involved is within the AtomGroup (`outside=False`), or whether *at least one* atom involved is within the AtomGroup (`outside=True`)

Reviewer 3: ```suggestion
- the [get_connections][] method to get bonds, angles, dihedrals and impropers in an AtomGroup, with the ability to specify whether all atoms involved are within the AtomGroup (`outside=False`), or whether *at least one* atom involved is outside the AtomGroup (`outside=True`)
```

Reviewer 3: (changed within -> outside because that seems like it would be the intention, but I haven't actually looked into `get_connections` so may have misunderstood)

Reviewer 2: Yeah the difference is:

`ag.get_connections("bonds", outside=False)`: both atom a and atom b of the bond are in `ag`

`ag.get_connections("bonds", outside=True)`: at least one of atom a and atom b is *inside* `ag`, but it does not guarantee that at least one atom is *outside*

Author: So my take here is the first is -> are is fine, but the second should be `within`?

Notable changes to analysis include:
- The argument order to AnalysisFromFunction are now as specified in the documentation
- The `select` keyword has been standardized by removing `selection`, `atomselection`, and `ref_select` in the `contact`, `gnm`, `helanal`, `hole`, `encore`, and `hydrogenbonds` modules
- The `save()` functions have been removed from `contacts`, `diffusionmap`, `hole`, `lineardensity`, and `rms` modules

> - `PCA(align=True)` now correctly aligns the trajectory

Author: Should we link to the issue here? This is the type of thing I would expect folks to want to know how it affected their analyses.

Reviewer: good idea

Reviewer: On that note: link to issues for neighbor searching? After all, NF paid @richardjgowers to fix them (small dev grant).

Reviewer 3: There might be a more elegant way to link the issues, but I've added something for now

> - Specifying `n_components` now correctly selects the PCA components

Author: As above.

- Progress bars have been replaced with an improved version from tqdm

> - The `radius_cut_q` method has been added to [contacts.Contacts](https://docs.mdanalysis.org/stable/documentation_pages/analysis/contacts.html#MDAnalysis.analysis.contacts.Contacts)

Author: ```suggestion
- The `radius_cut_q` method has been added to [contacts.Contacts](https://docs.mdanalysis.org/1.1.1/documentation_pages/analysis/contacts.html#MDAnalysis.analysis.contacts.Contacts)
```

> - `PCA(align=True)` now correctly aligns the trajectory

Reviewer 3: ```suggestion
- `PCA(align=True)` now correctly aligns the trajectory (See [Issue 2510](https://github.com/MDAnalysis/mdanalysis/issues/2510))
```

> - Specifying `n_components` now correctly selects the PCA components

Reviewer 3: ```suggestion
- Specifying `n_components` now correctly selects the PCA components (See [Issue 2623](https://github.com/MDAnalysis/mdanalysis/issues/2623))
```

- The notwithin_coordinates_factory() and density_from_PDB() methods of analysis.density are deprecated.
- analysis.waterdynamics.HydrogenBondLifetimes is deprecated in favor of analysis.hydrogenbonds.hbond_analysis.HydrogenBondAnalysis.lifetime() (to be implemented in version 2.0.0)
- analysis.leaflets.LeafletFinder() will no longer accept a filename, in 2.0.0 only Universes will be supported as inputs.
- analysis.helanal is deprecated and will be replaced by analysis.helix_analysis in 2.0.0.
- analysis.hbonds.WaterBridgeAnalysis will be moved to analysis.hydrogenbonds.WaterBridgeAnalysis.

The following parts of the readers/writers will be removed/changed in version 2.0.0:
- Writer.write_next_timestep() is deprecated in favor of Writer.write().
- Passing Timestep objects to Writer.write() is deprecated. In 2.0.0 only Universe or AtomGroup objects will be accepted.
- The way in which the NCDFWriter handles scale factos will change in version 2.x (see Issue #2327 for more details).
- When writing PDB files, MDAnalysis will no longer be using the last letter of the SegID to set the chainID in version 2.0.0.

> - The bfactors and tempfactors attributes (set by the PDB and MMTF parsers respectively), will be aliased in version 2.0.0.

Author: ```suggestion
- The `bfactors` and `tempfactors` attributes (set by the PDB and MMTF parsers respectively), will be aliased in version 2.0.0.
```

- When parsing TPR files, resids will be indexed from 1 rather than the current default of 0.

The following part of the core and library components will be removed/changed in version 2.0.0:

> - lib.log.echo() is deprecated in favor of the new lig.log.ProgressBar.

Reviewer: ```suggestion
 lib.log.echo() is deprecated in favor of the new lib.log.ProgressBar.
```

and featured the work of 25 new contributors:

 - [@bdice](https://github.com/bdice)
 - [@joaomcteixeira](https://github.com/joaomcteixeira)
 - [@CCook96](https://github.com/CCook96)
 - [@Yuan-Yu](https://github.com/Yuan-Yu)
 - [@HTian1997](https://github.com/HTian1997)
 - [@Iv-Hristov](https://github.com/Iv-Hristov)
 - [@hmacdope](https://github.com/hmacdope)
 - [@AnshulAngaria](https://github.com/AnshulAngaria)
 - [@ss62171](https://github.com/ss62171)
 - [@Luthaf](https://github.com/Luthaf)
 - [@yuxuanzhuang](https://github.com/yuxuanzhuang)
 - [@abhishandy](https://github.com/abhishandy)
 - [@mlnance](https://github.com/mlnance)
 - [@shfrz](https://github.com/shfrz)
 - [@wvandertoorn](https://github.com/wvandertoorn)
 - [@cbouy](https://github.com/cbouy)
 - [@AmeyaHarmalkar](https://github.com/AmeyaHarmalkar)
 - [@Oscuro-Phoenix](https://github.com/Oscuro-Phoenix)
 - [@andrrizzi](https://github.com/andrrizzi)
 - [@WG150](https://github.com/WG150)
 - [@Marcello-Sega](https://github.com/Marcello-Sega)
 - [@calcraven](https://github.com/calcraven)
 - [@ianmkenney](https://github.com/ianmkenney)
 - [@rcrehuet](https://github.com/rcrehuet)

>  - [@hanatok](https://github.com/hanatok)

Reviewer: Someone did a lot of extra work here — you can just write @orbeckst and it will automatically link to the GH profile; this is part of the GH packages that parse the markdown.

## Upgrading to MDAnalysis version 1.1.1

To get all these features run either 

```bash
conda update -c conda-forge mdanalysis
```

or 
```bash
pip install --upgrade MDAnalysis

> ```

Reviewer: I would move the upgrade info closer to the top.

They are provided as a string for easy use with conda or pip:

 - Python 2.7: biopython==1.76 cython==0.29.15 griddataformats==0.5.0 gsd==1.7.0 matplotlib==2.2.5 mmtf-python==1.1.2 netcdf4==1.3.1 numpy==1.16.5 scipy==1.2.1 tqdm==4.60.0
 - Python 3.5: biopython==1.72 cython==0.28.5 griddataformats==0.5.0 gsd==1.5.3 matplotlib==3.0.0 mmtf-python==1.1.2 netcdf4==1.3.1 numpy==1.15.2 scipy==1.1.0 tqdm==4.60.0
 - Python 3.6: biopython==1.78 cython==0.29.23 griddataformats==0.5.0 gsd==2.1.2 matplotlib==3.3.2 mmtf-python==1.1.2 netcdf4==1.5.4 numpy==1.16.0 scipy==1.5.1 tqdm==4.60.0
 - Python 3.7: biopython==1.78 cython==0.29.23 griddataformats==0.5.0 gsd==2.4.2 matplotlib==3.4.1 mmtf-python==1.1.2 netcdf4==1.5.6 numpy==1.20.2 scipy==1.6.3 tqdm==4.60.0
 - Python 3.8: biopython==1.78 cython==0.29.23 griddataformats==0.5.0 gsd==2.4.2 matplotlib==3.4.1 mmtf-python==1.1.2 netcdf4==1.5.6 numpy==1.20.2 scipy==1.6.3 tqdm==4.60.0

— The MDAnalysis Team

[roadmap]: {{ site.baseurl }}{% post_url 2019-11-06-roadmap %}
[interoperability]: {{ site.baseurl }}{% post_url 2020-08-03-roadmap %}
[CHANGELOG]: https://github.com/MDAnalysis/mdanalysis/blob/release-1.1.1/package/CHANGELOG

> [add_bonds]: https://docs.mdanalysis.org/stable/documentation_pages/core/universe.html#MDAnalysis.core.universe.Universe.add_bonds

Author: ```suggestion
[add_bonds]: https://docs.mdanalysis.org/1.1.1/documentation_pages/core/universe.html#MDAnalysis.core.universe.Universe.add_bonds
```

> [add_angles]: https://docs.mdanalysis.org/stable/documentation_pages/core/universe.html#MDAnalysis.core.universe.Universe.add_angles

Author: ```suggestion
[add_angles]: https://docs.mdanalysis.org/1.1.1/documentation_pages/core/universe.html#MDAnalysis.core.universe.Universe.add_angles
```

> [add_dihedrals]: https://docs.mdanalysis.org/stable/documentation_pages/core/universe.html#MDAnalysis.core.universe.Universe.add_dihedrals

Author: ```suggestion
[add_dihedrals]: https://docs.mdanalysis.org/1.1.1/documentation_pages/core/universe.html#MDAnalysis.core.universe.Universe.add_dihedrals
```

> [add_impropers]: https://docs.mdanalysis.org/stable/documentation_pages/core/universe.html#MDAnalysis.core.universe.Universe.add_impropers

Author: ```suggestion
[add_impropers]: https://docs.mdanalysis.org/1.1.1/documentation_pages/core/universe.html#MDAnalysis.core.universe.Universe.add_impropers
```

> [del_bonds]: https://docs.mdanalysis.org/stable/documentation_pages/core/universe.html#MDAnalysis.core.universe.Universe.delete_bonds

Author: ```suggestion
[del_bonds]: https://docs.mdanalysis.org/1.1.1/documentation_pages/core/universe.html#MDAnalysis.core.universe.Universe.delete_bonds
```

> [del_angles]: https://docs.mdanalysis.org/stable/documentation_pages/core/universe.html#MDAnalysis.core.universe.Universe.delete_angles

Author: ```suggestion
[del_angles]: https://docs.mdanalysis.org/1.1.1/documentation_pages/core/universe.html#MDAnalysis.core.universe.Universe.delete_angles
```

> [del_dihedrals]: https://docs.mdanalysis.org/stable/documentation_pages/core/universe.html#MDAnalysis.core.universe.Universe.delete_dihedrals

Author: ```suggestion
[del_dihedrals]: https://docs.mdanalysis.org/1.1.1/documentation_pages/core/universe.html#MDAnalysis.core.universe.Universe.delete_dihedrals
```

> [del_impropers]: https://docs.mdanalysis.org/stable/documentation_pages/core/universe.html#MDAnalysis.core.universe.Universe.delete_impropers

Author: ```suggestion
[del_impropers]: https://docs.mdanalysis.org/1.1.1/documentation_pages/core/universe.html#MDAnalysis.core.universe.Universe.delete_impropers
```

> [get_connections]: https://docs.mdanalysis.org/stable/documentation_pages/core/groups.html#MDAnalysis.core.groups.AtomGroup.get_connections

Author: ```suggestion
[get_connections]: https://docs.mdanalysis.org/1.1.1/documentation_pages/core/groups.html#MDAnalysis.core.groups.AtomGroup.get_connections
```

- a [DensityAnalysis](https://docs.mdanalysis.org/1.1.1/documentation_pages/analysis/density.html#MDAnalysis.analysis.density.DensityAnalysis) class for improved density analysis, replacing the now deprecated [density_from_Universe()](https://docs.mdanalysis.org/1.1.1/documentation_pages/analysis/density.html#MDAnalysis.analysis.density.density_from_Universe) code
- a method to compute the root-mean-square-inner-product of subspaces
- a method to calculate the cumulative overlap of a vector in a subspace

Other additions to core functionality include:
- [wrap and unwrap on-the-fly transformations](https://docs.mdanalysis.org/1.1.1/documentation_pages/transformations/wrap.html)
- methods to add [bonds][add_bonds], [angles][add_angles], [dihedrals][add_dihedrals], and [impropers][add_impropers] to a Universe
- methods to delete [bonds][del_bonds], [angles][del_angles], [dihedrals][del_dihedrals], and [impropers][del_impropers] from a Universe

> - the [get_connections][] method to get bonds, angles, dihedrals and impropers in an AtomGroup, with the ability to specify whether all atoms involved is within the AtomGroup (`outside=False`), or whether *at least one* atom involved is within the AtomGroup (`outside=True`)

Author: ```suggestion
- the [get_connections][] method to get bonds, angles, dihedrals and impropers in an AtomGroup, with the ability to specify whether all atoms involved are within the AtomGroup (`outside=False`), or whether *at least one* atom involved is within the AtomGroup (`outside=True`)
```

Miscellaneous performance improvements include:
- Dihedral selection in the Ramachandran class has been sped up ~700x.
- TPR parsing has been sped up 2–30x.

## Notable improvements

We have improved the flexibility to our atom selection language, allowing for advanced pattern matching operators.

> For example, we now support the `?` for single character matching, so using `T?R` in a selection string for a protein would yield both `THR` and TYR`. More information can be found in our [selection documentation](https://docs.mdanalysis.org/1.1.1/documentation_pages/selections.html#pattern-matching).

Author: ```suggestion
For example, we now support `?` for single character matching, so using `resname T?R` in a selection string for a protein would yield both residues `THR` and `TYR`. More information can be found in our [selection documentation](https://docs.mdanalysis.org/1.1.1/documentation_pages/selections.html#pattern-matching).
```

Other notable improvements to the core functionality include:
- AtomGroup.guess_bonds now uses periodic boundary information when available
- The TPRParser now supports GROMACS 2020
- When reading PDB and XYZ files, MDAnalysis now adds an `elements` attribute if the provided elements are valid

## Important fixes

For the full list of fixed please see the [CHANGELOG][]. The following are selection of fixes that could have either lead to wrong results or were often reported by users as problematic:

### Core

- Neighbor searching, which is a fundamental component of many analyses in MDAnalysis (such as hydrogenbonds and RDF calculation) had a number of bugs in 1.0.0 that could lead to wrong results, in particular with triclinic unit cells. The buggy code was disabled in 1.0.1 and fixed in 1.1.1. See issues [#2229][], [#2345][], [#2919][], [#2670][] [#2930][] for details.
- Fixed a SegmentationFault for the selection "around 0.0 SELECTION" (Issue [#2656][])
- `AtomGroup.center()` now works correctly for compounds and unwrapping (Issue [#2984][])

> - When bonds are guessed from distances (`AtomGroup.guess_bonds`), periodic boundary information is properly taken into account. Bonds that were split across the periodic boundary would have not beend correctly guessed previously. (Issue [#2350][])

Author: Repeat from line 96? is this intentional / do we want to keep both entries?
